### PR TITLE
feat(rule-engine): list rule support order by timestamp

### PR DESCRIPTION
**Pull request by wwhai**
Fixes <issue-number>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information

### CODE REVIEW

File: apps/emqx_rule_engine/src/emqx_rule_engine_api.erl [hunk 1]
```diff
@@ -495,8 +495,8 @@ parse_rule_params([{<<"actions">>, Actions} | Params], Rule) ->
     parse_rule_params(Params, Rule#{actions => parse_actions(Actions)});
 parse_rule_params([{<<"description">>, Descr} | Params], Rule) ->
     parse_rule_params(Params, Rule#{description => Descr});
-parse_rule_params([_ | Params], Res) ->
-    parse_rule_params(Params, Res).
+parse_rule_params([_ | Params], Rule) ->
+    parse_rule_params(Params, Rule#{created_at => erlang:system_time(millisecond)}).
```
Comment (terry-xiaoyu) [inline apps/emqx_rule_engine/src/emqx_rule_engine_api.erl:499]: create_at should be calculated in the emqx_rule_engine:create_rule/1.

File: apps/emqx_rule_engine/src/emqx_rule_registry.erl [hunk 2]
```diff
@@ -168,6 +170,14 @@ start_link() ->
 get_rules() ->
     get_all_records(?RULE_TAB).
 
+get_rules_order_by_ts() ->
```
Comment (terry-xiaoyu) [inline apps/emqx_rule_engine/src/emqx_rule_registry.erl:173]: get_rules_ordered_by_ts

Comment (terry-xiaoyu) [inline apps/emqx_rule_engine/src/emqx_rule_registry.erl:173]: The function get_rules_ordered_by_ts/0 should be used both by API and CLI, and the test case for the API and CLI should be updated to verify the ordering.

**Review by terry-xiaoyu — CHANGES_REQUESTED** [review]

File: apps/emqx_rule_engine/src/emqx_rule_registry.erl [hunk 3]
```diff
@@ -168,6 +170,14 @@ start_link() ->
 get_rules() ->
     get_all_records(?RULE_TAB).
 
+get_rules_ordered_by_ts() ->
+    F = fun() ->
+        Query = qlc:q([E || E <- mnesia:table(?RULE_TAB)]),
+        qlc:e(qlc:keysort(#rule.created_at, Query, [{order, ascending}]))
+    end,
+    {atomic, List} = mnesia:transaction(F),
+    List.
```
Comment (terry-xiaoyu) [inline apps/emqx_rule_engine/src/emqx_rule_registry.erl:179]: We need to add a test case for this function.

**Review by terry-xiaoyu — APPROVED** [review]

**Review by terry-xiaoyu — APPROVED** [review]

**Review by HJianBo — APPROVED** [review]

**Comment by HJianBo** [comment]
Does CLI need to return an ordered rules?

**Pull request merged**